src/time.cpp

```diff
@@ -69,6 +71,28 @@ namespace g3 {
    /// * format string must conform to std::put_time
    /// This is similar to std::put_time(std::localtime(std::time_t*), time_format.c_str());
 
+   namespace {
+      std::string kNanoSecondIdentifier = "%f";
+   }
+
+   std::string localtime_formatted(const timespec &time_snapshot, const std::string &time_format) {
+      auto format_buffer = time_format;
+      auto value_str = std::to_string(time_snapshot.tv_nsec);
+
+      // creating nsec string with leading zeros
+      auto nsec_str = std::string("000000000");
+      nsec_str.replace(nsec_str.length() - value_str.length(), value_str.length(), value_str);
```

Comment:
I have no particular opinion whether or not leading zeroes is a good thing or not. 

You can simplify this, potentially speed it up?  by creating one string less
```
std::string ns = std::string(number_zeroes - ns_value.length(), '0') + ns_value;
```
I.e. pretty much what you already wrote, only simplified.